magento/pwa-studio
# Pwa 2172: Validate Shopping Cart Promotions are displayed correctly

## Description

As a merchant I want to be able to personalize promotions through segmentation so that I can provide the best possible experience to my customers

Acceptance Criteria

Shopping cart price rules should be applied to relevant carts accurately
Cart price rules based on segmentation must reflect the accurate customer segment
Display Shopping Cart Price Rule Label in cart summary

Example Scenarios

Free Shipping > $50 cart total, no exclusions
Free Shipping for Registered Customers only - registered customer segment applied to rule
10% Discount over $500 Cart value for Guest Customers - guest customer segment applied to rule, not applicable for registered users
15% Discount over $500 Cart value for Registered Customers - registered customer segment applied to rule, not applicable for registered users
20% Discount for high spending customers - registered customer segment, total historical order value > $1000

## Related Issue







Closes #[Pwa 2172](https://jira.corp.magento.com/browse/PWA-2172)

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps





Test Plan
Execute -

Verify cart price rules are displayed based on how they are configured in backend.
Verify all scenarios from the example.
Verify Merge cart flow from guest to registered user where guest cart has some rule applied and registered user cart has something different.
Test Coverage -

Add cypress integration test coverage.

#### Test scenario(s) for direct fix/feature





#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
